R/rdeephaven/R/table_handle_wrapper.R

```diff
@@ -0,0 +1,48 @@
+TableHandle <- R6Class("TableHandle",
+    public = list(
+
+        initialize = function(table_handle) {
+
+            # first, type checking
+            if (class(table_handle)[[1]] != "Rcpp_INTERNAL_TableHandle") {
+                stop("'table_handle' should be an internal Deephaven TableHandle. If you're seeing this,
+                you are trying to call the constructor of TableHandle directly, which is not advised.")
+            }
+            private$internal_table_handle <- table_handle
+        },
+
+        bind_to_variable = function(name) {
+            if (class(name)[[1]] != "character") {
+                stop("'name' should be a character or a string.")
+            }
+            private$internal_table_handle$bind_to_variable(name)
+        },
+
+        to_arrow_record_batch_stream_reader = function() {
+            ptr = private$internal_table_handle$get_arrow_array_stream_ptr()
+            rbsr = RecordBatchStreamReader$import_from_c(ptr)
+            return(rbsr)
+        },
+
+        to_arrow_table = function() {
+            rbsr = self$to_arrow_record_batch_stream_reader()
+            arrow_tbl = rbsr$read_table()
+            return(arrow_tbl)
+        },
+
+        to_tibble = function() {
+            rbsr = self$to_arrow_record_batch_stream_reader()
+            arrow_tbl = rbsr$read_table()
+            return(as_tibble(arrow_tbl))
+        },
+
+        to_data_frame = function() {
+            arrow_tbl = self$to_arrow_table()
+            return(as.data.frame(as.data.frame(arrow_tbl))) # for some reason as.data.frame on arrow table returns a tibble, not what I want
```

Comment:
This looks wrong, as I noted in the other comment that appears closed.

Also "I" , "we", "us" etc should not be in the comments.  You are developing in public, so all comments need to be "professional" for the project.

Reply:
I don't think I closed the comment, but I did comment and mention that this is here because as.data.frame on an arrow_table returns a tibble rather than a pure data frame.

`> static_table <- client$open_table("static_table1")`

`> data_frame <- static_table$to_data_frame()`
`> class(data_frame)`
`[1] "data.frame"`

`> arrow_table <- static_table$to_arrow_table()`
`> class(arrow_table)`
`[1] "Table"        "ArrowTabular" "ArrowObject"  "R6"`

`> tibbl <- static_table$to_tibble()`
`> class(tibbl)`
`[1] "tbl_df"     "tbl"        "data.frame"`

`> data_frame_from_arrow_table = as.data.frame(arrow_table)`
`> class(data_frame_from_arrow_table)`
`[1] "tbl_df"     "tbl"        "data.frame"`

`> new_data_frame_from_arrow_table = as.data.frame(as.data.frame(arrow_table))`
`> class(new_data_frame_from_arrow_table)`
`[1] "data.frame"`

I think this probably has more to do with R being weird about inheritance than someone being outright wrong. I can remove the second as.data.frame, but the result in that case is not what I expect.

Reply:
The big question is if it is allocating more memory or not.  If not, it doesn't matter.  If so, then it subpar but can be addressed in v2.